integration/apps/pennant/pennant.py

```diff
@@ -0,0 +1,93 @@
+#  modification, are permitted provided that the following conditions
+#  are met:
+#
+#      * Redistributions of source code must retain the above copyright
+#        notice, this list of conditions and the following disclaimer.
+#
+#      * Redistributions in binary form must reproduce the above copyright
+#        notice, this list of conditions and the following disclaimer in
+#        the documentation and/or other materials provided with the
+#        distribution.
+#
+#      * Neither the name of Intel Corporation nor the names of its
+#        contributors may be used to endorse or promote products derived
+#        from this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+#  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+#  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+#  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+#  OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+#  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+#  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+#  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+#  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+#  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY LOG OF THE USE
+#  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+'''
+AppConf class for Pennant benchmark.
+'''
+
+import os
+import re
+
+from apps import apps
+
+
+def setup_run_args(parser):
+    """ Add common arguments for all run scripts:
+        --input
+    """
+    parser.add_argument('--input', "-i",
+                        action='store', type=str,
+                        default="PENNANT/test/leblancx4/leblancx4.pnt",
+                        help='Path to the input file (see .pnt files in test' +
+                             'directory of the PENNANT source tarball). ' +
+                             'Absolute path or relative to the app ' +
+                             'directory. Default is nohpoly.pnt')
+
+
+class PennantAppConf(apps.AppConf):
+    @staticmethod
+    def name():
+        return 'pennant'
+
+    def __init__(self, problem_file="PENNANT/test/leblancx4/leblancx4.pnt"):
+        benchmark_dir = os.path.dirname(os.path.abspath(__file__))
+        self._exec_path = os.path.join(benchmark_dir, 'PENNANT/build/pennant')
+        if os.path.isfile(problem_file):
+            self._exec_args = problem_file
+        elif os.path.isfile(os.path.join(benchmark_dir, problem_file)):
+            self._exec_args = os.path.join(benchmark_dir, problem_file)
+        else:
+            raise RuntimeError("Input file not found: " + problem_file)
+
+    def get_rank_per_node(self):
+        return 2
+
+    def get_cpu_per_rank(self):
+        return 1
```

Comment:
This still looks wrong to me.  2 ranks per node and 1 cpu per rank means that there will be 2 cpus (i.e. logical cores) used in total.  That seems to be drastically underutilizing the 44 core (2 sockets * 22 cores) nodes we have.

If you leave this unset, the launcher will try to use as many cores as it can to fill up the sockets.

Reply:
Oh shoot, sorry it was 42 and I had changed it to 2 to debug the error. I will change it back.

Reply:
Fixed.

Reply:
OK, this looks fine for now.

Once testing starts on machines that are not McFly, this file will need to be updated to use the machine file to probe for how many cores there are, and then set this appropriately.